source/gameengine/Rasterizer/RAS_CubeMap.cpp

```diff
@@ -0,0 +1,197 @@
+* modify it under the terms of the GNU General Public License
+* as published by the Free Software Foundation; either version 2
+* of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*
+* Contributor(s): Ulysse Martin, Tristan Porteries.
+*
+* ***** END GPL LICENSE BLOCK *****
+*/
+
+/** \file RAS_CubeMap.cpp
+*  \ingroup bgerast
+*/
+
+#include "RAS_CubeMap.h"
+#include "RAS_Texture.h"
+#include "RAS_IRasterizer.h"
+
+#include "GPU_texture.h"
+#include "GPU_framebuffer.h"
+#include "GPU_draw.h"
+
+#include "DNA_texture_types.h"
+
+#include "glew-mx.h"
+
+static const MT_Matrix3x3 topFaceViewMat(
+	1.0f, 0.0f, 0.0f,
+	0.0f, -1.0f, 0.0f,
+	0.0f, 0.0f, -1.0f);
+
+static const MT_Matrix3x3 bottomFaceViewMat(
+	-1.0f, 0.0f, 0.0f,
+	0.0f, -1.0f, 0.0f,
+	0.0f, 0.0f, 1.0f);
+
+static const MT_Matrix3x3 frontFaceViewMat(
+	0.0f, 0.0f, -1.0f,
+	0.0f, -1.0f, 0.0f,
+	-1.0f, 0.0f, 0.0f);
+
+static const MT_Matrix3x3 backFaceViewMat(
+	0.0f, 0.0f, 1.0f,
+	0.0f, -1.0f, 0.0f,
+	1.0f, 0.0f, 0.0f);
+
+static const MT_Matrix3x3 rightFaceViewMat(
+	1.0f, 0.0f, 0.0f,
+	0.0f, 0.0f, -1.0f,
+	0.0f, 1.0f, 0.0f);
+
+static const MT_Matrix3x3 leftFaceViewMat(
+	1.0f, 0.0f, 0.0f,
+	0.0f, 0.0f, 1.0f,
+	0.0f, -1.0f, 0.0f);
+
+const MT_Matrix3x3 RAS_CubeMap::faceViewMatrices3x3[RAS_CubeMap::NUM_FACES] = {
+	topFaceViewMat,
+	bottomFaceViewMat,
+	frontFaceViewMat,
+	backFaceViewMat,
+	rightFaceViewMat,
+	leftFaceViewMat
+};
+
+static const GLenum cubeMapTargets[RAS_CubeMap::NUM_FACES] = {
+	GL_TEXTURE_CUBE_MAP_POSITIVE_Z_ARB,
+	GL_TEXTURE_CUBE_MAP_NEGATIVE_Z_ARB,
+	GL_TEXTURE_CUBE_MAP_POSITIVE_X_ARB,
+	GL_TEXTURE_CUBE_MAP_NEGATIVE_X_ARB,
+	GL_TEXTURE_CUBE_MAP_POSITIVE_Y_ARB,
+	GL_TEXTURE_CUBE_MAP_NEGATIVE_Y_ARB,
+};
+
+RAS_CubeMap::RAS_CubeMap(RAS_Texture *texture)
+	:m_gpuTex(NULL),
+	m_useMipmap(false),
+	m_texture(texture)
+{
+	for (unsigned short i = 0; i < NUM_FACES; ++i) {
+		m_fbos[i] = NULL;
+		m_rbs[i] = NULL;
+	}
+	GetValidTexture();
+}
+
+RAS_CubeMap::~RAS_CubeMap()
+{
+	DetachTexture();
+
+	/* This call has for side effect to ask regeneration of all textures
+	 * depending of this image.
+	 */
+	GPU_free_image(m_texture->GetImage());
+}
+
+void RAS_CubeMap::AttachTexture()
+{
+	// Increment reference to make sure the gpu texture will not be freed by someone else.
+	GPU_texture_ref(m_gpuTex);
+
+	for (unsigned short i = 0; i < NUM_FACES; ++i) {
+		m_fbos[i] = GPU_framebuffer_create();
+		m_rbs[i] = GPU_renderbuffer_create(GPU_texture_width(m_gpuTex), GPU_texture_height(m_gpuTex),
```

Comment:
Hi! Do we really need renderbuffers? In most examples I see on the net for cubemaps there is no renderbuffer.